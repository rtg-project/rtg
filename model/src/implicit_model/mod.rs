pub use crate::explicit_model::entity::ImplicitEntity;
pub use crate::explicit_model::field::ImplicitField;
pub use crate::explicit_model::model::ImplicitModel;
pub use crate::explicit_model::sql_type::Type;

pub mod json_schema;

// Tests
#[cfg(test)]
mod tests {
  use super::*;
  use insta::assert_json_snapshot;
  use std::rc::Rc;

  #[test]
  fn serialize() {
    let value = ImplicitModel::V1 {
      entities: Some(vec![Rc::new(ImplicitEntity::DatabaseTable {
        name: Some("person".to_string()),
        sql_schema_name: Some("public".to_string()),
        sql_table_name: Some("person_table".to_string()),
        graphql_entity_type_name: Some("Person".to_string()),
        graphql_filter_type_name: Some("PersonWhereFilter".to_string()),
        graphql_get_single_operation_name: None,
        graphql_get_list_operation_name: None,
        graphql_get_connection_operation_name: Some("personConnection".to_string()),
        graphql_default_order_by: Some("id_ASC".to_string()),
        graphql_default_first: Some(10),
        graphql_default_offset: Some(0),
        fields: Some(vec![
          Rc::new(ImplicitField::ScalarDatabaseColumn {
            name: Some("id".to_string()),
            sql_type: Some(Type::Text),
            sql_column_name: Some("id_col".to_string()),
            graphql_field_name: Some("id".to_string()),
            graphql_type_name: Some("String".to_string()),
            graphql_order_by_asc: None,
            graphql_order_by_desc: None,
          }),
          Rc::new(ImplicitField::ScalarDatabaseColumn {
            name: Some("drone".to_string()),
            sql_type: Some(Type::Text),
            sql_column_name: Some("drone_col".to_string()),
            graphql_field_name: Some("drone".to_string()),
            graphql_type_name: Some("String".to_string()),
            graphql_order_by_asc: Some("drone_ASC".to_string()),
            graphql_order_by_desc: Some("drone_DESC".to_string()),
          }),
        ]),
      })]),
    };

    assert_json_snapshot!(serde_json::to_string_pretty(&value).unwrap());
  }
}
