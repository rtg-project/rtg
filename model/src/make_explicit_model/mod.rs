pub mod conversion_error;
pub mod convert_entity;
pub mod convert_field;
pub mod convert_model;

// Tests
#[cfg(test)]
mod tests {

  use super::*;
  use crate::implicit_model::*;
  use insta::assert_debug_snapshot;
  use std::rc::Rc;

  #[test]
  fn it_works() {
    let value = ImplicitModel::V1 {
      entities: Some(vec![Rc::new(ImplicitEntity::DatabaseTable {
        name: Some("person".to_owned()),
        sql_schema_name: Some("public".to_owned()),
        sql_table_name: Some("person_table".to_owned()),
        graphql_entity_type_name: Some("Person".to_owned()),
        graphql_filter_type_name: Some("PersonWhereFilter".to_owned()),
        graphql_get_single_operation_name: None,
        graphql_get_list_operation_name: None,
        graphql_get_connection_operation_name: None,
        graphql_default_order_by: Some("id_ASC".to_owned()),
        graphql_default_first: Some(10),
        graphql_default_offset: Some(0),
        fields: Some(vec![
          Rc::new(ImplicitField::ScalarDatabaseColumn {
            name: Some("id".to_owned()),
            nullable: Some(false),
            sql_type: Some(Type::Text),
            sql_column_name: Some("id_col".to_owned()),
            graphql_field_name: Some("id".to_owned()),
            graphql_type_name: Some("ID".to_owned()),
            graphql_order_by_asc: None,
            graphql_order_by_desc: None,
          }),
          Rc::new(ImplicitField::ScalarDatabaseColumn {
            name: Some("drone".to_owned()),
            nullable: Some(false),
            sql_type: Some(Type::Text),
            sql_column_name: None,
            graphql_field_name: None,
            graphql_type_name: Some("Integer".to_owned()),
            graphql_order_by_asc: Some("drone_ASC".to_owned()),
            graphql_order_by_desc: Some("drone_DESC".to_owned()),
          }),
        ]),
      })]),
    };

    assert_debug_snapshot!(convert_model::convert_model(&value));
  }
}
