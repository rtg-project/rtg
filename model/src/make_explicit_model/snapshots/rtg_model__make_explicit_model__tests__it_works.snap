---
source: model/src/make_explicit_model/mod.rs
expression: "convert_model::convert_model(&value)"

---
Ok(
    V1 {
        entities: [
            DatabaseTable {
                name: "person",
                sql_schema_name: "public",
                sql_table_name: "person_table",
                graphql_entity_type_name: "Person",
                graphql_filter_type_name: "PersonWhereFilter",
                graphql_get_single_operation_name: "getPerson",
                graphql_get_list_operation_name: "allPersonople",
                graphql_get_connection_operation_name: "allPersonopleConnection",
                graphql_default_order_by: "id_ASC",
                graphql_default_first: 10,
                graphql_default_offset: 0,
                fields: [
                    ScalarDatabaseColumn {
                        name: "id",
                        nullable: false,
                        sql_type: Uuid,
                        sql_column_name: "id_col",
                        graphql_enabled: true,
                        graphql_field_name: "id",
                        graphql_type_name: "ID",
                        graphql_order_by_asc: "idAsc",
                        graphql_order_by_desc: "idDesc",
                    },
                    ScalarDatabaseColumn {
                        name: "drone",
                        nullable: false,
                        sql_type: Text,
                        sql_column_name: "drone",
                        graphql_enabled: true,
                        graphql_field_name: "drone",
                        graphql_type_name: "String",
                        graphql_order_by_asc: "drone_ASC",
                        graphql_order_by_desc: "drone_DESC",
                    },
                ],
            },
        ],
    },
)
